src/extensions/struct-conn/index.ts

```diff
@@ -0,0 +1,365 @@
+        }
+        const conns = structConnCache.get(structNodeRef)!;
+        const conn = conns[structConnId];
+        if (!conn) {
+            console.error(`The structure does not contain struct_conn "${structConnId}"`);
+            return;
+        }
+
+        if (!keepExisting) {
+            await removeAllStructConnInspections(plugin);
+        }
+        await addStructConnInspection(plugin, structNode, conn);
+        hideSnfgNodes(plugin);
+    },
+
+    /** Remove anything created by `inspectStructConn` and
+     * make visible any carbohydrate SNFG visuals that have been hidden by `inspectStructConn`.
+     */
+    async clearStructConnInspections(plugin: PluginContext) {
+        await removeAllStructConnInspections(plugin);
+        unhideSnfgNodes(plugin);
+    },
+};
+
+
+type StructNode = ReturnType<typeof selectStructureNode>
+
+/** Return the first structure node belonging to `entry` (e.g. '1tqn'),
+ * or to the first loaded structure node overall if `entry` is `undefined`.
+ * Includes only "root" structures, not structure components. */
+function selectStructureNode(plugin: PluginContext, entry: string | undefined) {
+    const structNodes = plugin.state.data
+        .selectQ(q => q.ofType(PluginStateObject.Molecule.Structure))
+        .filter(node => node.obj && !node.obj.data.parent && !node.transform.transformer.definition.isDecorator);
+    if (entry) {
+        const result = structNodes.find(node => node.obj && node.obj.data.model.entry.toLowerCase() === entry.toLowerCase());
+        if (!result) {
+            console.warn(`Structure with entry ID "${entry}" was not found. Available structures: ${structNodes.map(node => node.obj?.data.model.entry)}`);
+        }
+        return result;
+    } else {
+        if (structNodes.length > 1) {
+            console.warn(`Structure entry ID was not specified, but there is more than one loaded structure (${structNodes.map(node => node.obj?.data.model.entry)}). Taking the first structure.`);
+        }
+        if (structNodes.length === 0) {
+            console.warn(`There are no loaded structures.`);
+        }
+        return structNodes[0];
+    }
+}
+
+
+/** Represents one partner (i.e. atom) of a struct_conn */
+interface StructConnPartner {
+    asymId: string,
+    seqId: number | undefined,
+    authSeqId: number | undefined,
+    insCode: string,
+    compId: string,
+    atomId: string,
+    /** Alternative location (use empty string if not given) */
+    altId: string,
+}
+
+/** Represents a struct_conn (interaction between two partners) */
+export interface StructConnRecord {
+    id: string,
+    distance: number,
+    partner1: StructConnPartner,
+    partner2: StructConnPartner,
+}
+
+
+/** Return an object with all struct_conn records read from mmCIF.
+ * Return {} if structure comes from another format than mmCIF.
+ */
+function extractStructConns(structure: Structure): { [id: string]: StructConnRecord } {
```

Comment:
```suggestion
function extractStructConns(model: Model): { [id: string]: StructConnRecord } {
```

and cache the result in `model._staticPropertyData['wwpdb-struct-conn-extension-data']`. Then you don't need the plugin behavior.

Or just add `function getStructConn(model: Model, id: string)?: StructConn` and query that each time you need it. No need for cache then.

Reply:
> cache the result in model._staticPropertyData['wwpdb-struct-conn-extension-data']
I like this idea, but the `_staticPropertyData` doc says "Not to be accessed directly, each custom property descriptor defines property accessors that use this field to store the data." Should I just ignore it?

Reply:
Reality is not perfect. You can ignore this. The only issue this could cause if some other extension called the property `'wwpdb-struct-conn-extension-data'` too. See for example ``https://github.com/molstar/molstar/blob/master/src/extensions/pdbe/preferred-assembly.ts#L23` for how we use it.

Reply:
Should be fixed now.